packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -5802,8 +5802,9 @@ namespace Tpetra {
          << numPacketsPerLID.extent(0) << endl;
       std::cerr << os.str();
     }
+    const LO numPacketsPerLID_extent = numPacketsPerLID.extent (0);
     TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
-      (numExportLIDs != numPacketsPerLID.extent (0), std::runtime_error,
+      (numExportLIDs !=  numPacketsPerLID_extent, std::runtime_error,
```

Comment:
Just out of curiosity, what was the warning here? It looks like it was comparing one extent with another, so they should have both been ``size_t``...

Reply:
After I changed the type of numExportLIDs to be ``const LO`` I got the following error:

```
projects/albany/nightlyCDashTrilinosBlake/repos-gcc/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:5807:42: error: no match for call to '(Kokkos::DualView<long unsigned int*, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>, void, void>) (int)'
 5807 |       (numExportLIDs !=  numPacketsPerLID(0), std::runtime_error,
      |                          ~~~~~~~~~~~~~~~~^~~
/projects/albany/nightlyCDashTrilinosBlake/repos-gcc/Trilinos/packages/teuchos/core/src/Teuchos_TestForException.hpp:180:33: note: in definition of macro 'TEUCHOS_TEST_FOR_EXCEPTION'
  180 |   const bool throw_exception = (throw_exception_test); \
      |                                 ^~~~~~~~~~~~~~~~~~~~
/projects/albany/nightlyCDashTrilinosBlake/repos-gcc/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:5806:5: note: in expansion of macro 'TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC'
 5806 |     TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC
```

Reply:
I assume that should say ``(numExportLIDs != numPacketsPerLID.extent(0)`` and generate a -Wsign-compare, not an error?

Reply:
Sorry, in my hasty attempt to manually revert my change, I reverted it incorrectly.  The actual error is a sign comparison issue, yes: 

```
/projects/albany/nightlyCDashTrilinosBlake/repos-gcc/Trilinos/packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp:5807:22: error: comparison of integer expressions of different signedness: 'const LO' {aka 'const int'} and 'std::enable_if_t<true, long unsigned int>' {aka 'long unsigned int'} [-Werror=sign-compare]
```

Reply:
OK thanks. I would also be OK with just declaring ``numExportLIDs`` to be size_t instead, but it doesn't really matter